modules/cmdb/services/org/org.go

```diff
@@ -494,17 +494,19 @@ func (o *Org) GetOrgByDomain(domain string) (*model.Org, error) {
 	if domain != "" && conf.OrgWhiteList[domain] {
 		return nil, nil
 	}
-	suf := strutil.Concat(".", conf.RootDomain())
-	domain_and_port := strutil.Split(domain, ":", true)
-	domain = domain_and_port[0]
-	if !strutil.HasSuffixes(domain, suf) {
-		return nil, apierrors.ErrGetOrg.NotFound()
-	}
-	orgName := strutil.TrimSuffixes(domain, suf)
-	if strutil.HasSuffixes(orgName, "-org") {
-		orgName = strutil.TrimSuffixes(orgName, "-org")
+	for _, rootDomain := range conf.RootDomainList() {
```

Comment:
Split to a new method and add unit test please.

Reply:
done